packages/redux-dynamic-modules/src/DynamicModuleLoader.tsx

```diff
@@ -102,7 +102,30 @@ class DynamicModuleLoaderImpl extends React.Component<
             }
         }
 
-        this._addedModules = store.addModules(modules);
+        for(const module of modules) {
+            this._addedModules[module.id] = store.addModule(module);
+        }
+    }
+
+    public getSnapshotBeforeUpdate() {

```

Comment:
Isn't this too late? Like do we want to make sure if this component is re-rendered with additional modules that they are immediately available to children in the tree?

Reply:
@cschleiden@live.de Do you know of a lifecyle method where I can do this now?

---
In reply to: [246155351](https://github.com/Microsoft/redux-dynamic-modules/pull/43#discussion_r246155351) [](ancestors = 246155351)

Reply:
@abettadapur I couldn't find a good solution. A hacky one might be to put the store in the state and then use `getDerivedStateFromProps`, or do it in `render`, memoized in some way